includes/class-wp-security-bp-json.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * The file that defines the JSON array class
+ *
+ * This class handles all the files access to read and write.
+ *
+ * @link       http://example.com
+ * @since      1.0.0
+ *
+ * @package    wp_security_bp
+ * @subpackage wp_security_bp/includes
+ */
+
+/**
+ * The JSON class.
+ *
+ * This class handles all the information to be send by ajax response in an array.
+ *
+ * @since      1.0.0
+ * @package    wp_security_bp
+ * @subpackage wp_security_bp/includes
+ * @author     Cyan Lovers <hello@cyanlove.com>
+ */
+
+class WP_Security_BP_JSON {
+	/**
+	 * Final array to return formated for each case.
+	 *
+	 * @since    1.0.0
+	 * @access   protected
+	 * @var      array    $json    The array that will be returned to the admin (later transformed to JSON)
+	 */
+	public $json = array();
+
+	/**
+	 * The unique identifier of this plugin.
+	 *
+	 * @since    1.0.0
+	 * @access   protected
+	 * @var      string    $plugin_name    The string used to uniquely identify this plugin.
+	 */
+	protected $plugin_name;
+
+	/**
+	 * Initialize the class and set its properties.
+	 *
+	 * @since    1.0.0
+	 * @param    string    $plugin_name       The name of this plugin.
+	 */
+	public function __construct( $plugin_name ) {
+
+		$this->plugin_name = $plugin_name;
+		$this->json        = array(
+			'status'  => 'fail',
+			'message' => '',
+			'button'  => false,
+		);

```

Comment:
You still need the uri field! Just that it won’t be an uri, it will be an action. Also would be useful to have another field to identify the class that handles the action. Or maybe use a naming structure to handle the two fields in one string. 🤔